consensus/safety-rules/src/safety_rules.rs

```diff
@@ -103,7 +112,18 @@ impl SafetyRules {
         proposed_block
             .verify_well_formed()
             .map_err(|error| Error::InvalidProposal(error.to_string()))?;
-        self.extension_check(vote_proposal)
+        if self.decoupled_execution {
+            Ok(VoteData::new(
+                proposed_block.gen_block_info(
+                    *ACCUMULATOR_PLACEHOLDER_HASH,
+                    0,
+                    vote_proposal.next_epoch_state().cloned(),
+                ),
+                proposed_block.quorum_cert().certified_block().clone(),
+            ))
+        } else {
+            self.extension_check(vote_proposal)
```

Comment:
feels like we could add two methods to VoteProposal, vote_data and vote_data_with_extension_proof and eliminate this code from here?

Reply:
Thanks! This makes the code cleaner.